config/impl/src/main/java/org/jboss/windup/config/loader/RuleLoaderImpl.java

```diff
@@ -131,12 +132,24 @@ private void checkForDuplicateProviders(List<RuleProvider> providers)
 
     private RuleProviderRegistry build(GraphContext context, Predicate<RuleProvider> ruleProviderFilter)
     {
-
-        ConfigurationBuilder result = ConfigurationBuilder.begin();
+        List<Rule> allRules = new ArrayList<>(2000); // estimate of how many rules we will likely see
 
```

Comment:
It is just separating the creation of the Configuration from the rule list to allow the rule list to be configured independently.